go/vt/sqlparser/sql.y

```diff
@@ -938,10 +978,157 @@ value_expression:
   {
     $$ = $1
   }
+
 | VALUES openb sql_id closeb
   {
     $$ = &ValuesFuncExpr{Name: $3}
   }
+| CONVERT openb expression ',' convert_type closeb
+  {
+    $$ = &ConvertExpr{Expr: $3, Type: $5}
+  }
+| CAST openb expression AS convert_type closeb
+  {
+    $$ = &ConvertExpr{Expr: $3, Type: $5}
+  }
+| MATCH openb column_list closeb AGAINST openb value_expression match_option closeb
+  {
+  $$ = &MatchExpr{Columns: $3, Expr: $7, Option: $8}
+  }
+
+match_option:
+/*empty*/
+  {
+    $$ = ""
+  }
+| IN BOOLEAN MODE
+  {
+    $$ = BooleanModeStr
+  }
+| IN NATURAL LANGUAGE MODE
+ {
+    $$ = NaturalLanguageModeStr
+ }
+| IN NATURAL LANGUAGE MODE WITH QUERY EXPANSION
+ {
+    $$ = NaturalLanguageModeWithQueryExpansionStr
+ }
+| WITH QUERY EXPANSION
+ {
+    $$ = QueryExpansionStr
+ }
+
+
+convert_type:
+  BINARY openb INTEGRAL closeb
+  {
+    $$ = &ConvertType{Type: Binary, Length: NewIntVal($3)}
+  }
+| BINARY
+  {
+    $$ = &ConvertType{Type: Binary}
+  }
+| CHAR openb INTEGRAL closeb charset_info
+  {
+    $$ = &ConvertType{Type: Char, Length: NewIntVal($3), Charset: $5}
+  }
+| CHAR charset_info
+  {
+    $$ = &ConvertType{Type: Char, Charset: $2}
+  }
+| CHAR openb INTEGRAL closeb
+  {
+    $$ = &ConvertType{Type: Char, Length: NewIntVal($3)}
+  }
+| CHAR
+  {
+    $$ = &ConvertType{Type: Char}
+  }
+| NCHAR openb INTEGRAL closeb
+  {
+    $$ = &ConvertType{Type: NChar, Length: NewIntVal($3)}
+  }
+| NCHAR
+  {
+    $$ = &ConvertType{Type: NChar}
+  }
+| SIGNED integer_keyword_opt
+  {
+    $$ = &ConvertType{Type: Signed}
+  }
+| UNSIGNED integer_keyword_opt
+  {
+    $$ = &ConvertType{Type: Unsigned}
+  }
+| DECIMAL openb value ',' value closeb
+  {
+    if num1, ok := $3.(*SQLVal); ok && num1.Type == IntVal {
+      if num2, ok := $5.(*SQLVal); ok && num2.Type == IntVal {
+        $$ = &ConvertType{Type: Decimal, Length: num1, Scale: num2}
+      } else {
+        yylex.Error("expecting integer")
+        return 1
+      }
+    } else {
+      yylex.Error("expecting integer")
+      return 1
+    }
+  }
+| DECIMAL openb value closeb
+  {
+    if num, ok := $3.(*SQLVal); ok && num.Type == IntVal {
+      $$ = &ConvertType{Type: Decimal, Length: num}
+    } else {
+			yylex.Error("expecting integer")
+			return 1
+    }
+  }
+| DECIMAL
+  {
+    $$ = &ConvertType{Type: Decimal}
+  }
+| DATE
+  {
+    $$ = &ConvertType{Type: Date}
+  }
+| TIME openb INTEGRAL closeb
+  {
+    $$ = &ConvertType{Type: Time, Length: NewIntVal($3)}
+  }
+| TIME
+  {
+	  $$ = &ConvertType{Type: Time}
+  }
+| DATETIME openb INTEGRAL closeb
+  {
+	  $$ = &ConvertType{Type: Datetime, Length: NewIntVal($3)}
+  }
+| DATETIME
+  {
+	  $$ = &ConvertType{Type: Datetime}
+  }
+| JSON
+  {
+	  $$ = &ConvertType{Type: Json}
+  }
+
+integer_keyword_opt:
+  /*empty*/
+| INTEGER
+
+charset_info:
+  CHARACTER SET charset
+  {
+    $$ = $3
+  }
+| ASCII
+  {
+    $$ = "latin1"
```

Comment:
I think simplification would be better, mysql can handle the setting of real values, we mostly care about accurately re-representing the query on rewrite